### PR TITLE
Use missing_intervals for end

### PR DESCRIPTION
**Pull request by eakmanrq**

### CODE REVIEW

File: sqlmesh/core/plan/definition.py [hunk 1]
```diff
@@ -158,15 +158,29 @@ def set_start(self, new_start: TimeLike) -> None:
         self._start = new_start
         self.__missing_intervals = None
 
+    @property
+    def _default_end(self) -> TimeLike:
```
Comment (tobymao) [inline sqlmesh/core/plan/definition.py:162]: what's the purpose of this? can you delete this and override_end?

File: sqlmesh/core/plan/definition.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def __init__(
         self.categorizer_config = categorizer_config or CategorizerConfig()
         self.auto_categorization_enabled = auto_categorization_enabled
         self._start = start if start or not (is_dev and forward_only) else yesterday_ds()
-        self._end = end if end or not is_dev else now()
+        self._end = end
```
Comment (izeigerman) [inline sqlmesh/core/plan/definition.py:90]: Please note that no longer setting the end in the constructor impacts the end we set in the environment: https://github.com/TobikoData/sqlmesh/blob/48cc0be158c379f49cd72704cc1c3600cc23b27d/sqlmesh/core/plan/definition.py#L229. Please also note that we may expect end to be None in the environment in some cases.

**Comment by izeigerman** [comment]
Please note that this change will break Airflow here: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/schedulers/airflow/dag_generator.py#L503 and the built-in scheduler here - https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/plan/evaluator.py#L119

File: sqlmesh/core/plan/definition.py (hunk 1 above)
Comment (eakmanrq) [reply sqlmesh/core/plan/definition.py:162]: We refer to the default end in two places (that being `now()`) so I wanted to make sure they were consistent

**Review by izeigerman — APPROVED** [review]

**Pull request merged**